tests/cpp/ServerTests.cpp

```diff
@@ -510,87 +369,31 @@ BOOST_AUTO_TEST_CASE(testThreadedSmallSegmentStream)
 
             BOOST_CHECK(stream.finishFrame().get());
 
-            server->requestFrame(testStreamId);
+            requestFrame(testStreamId);
 
             // process frame receive
-            processServerMessages();
+            processServerMessage();
         }
     }
 
     // handle close of streamer
-    processServerMessages();
-
-    serverThread.quit();
-    serverThread.wait();
+    processServerMessage();
 
-    BOOST_CHECK_EQUAL(openedStreams, 0);
```

Comment:
I wasn't fully done yet yesterday, you review too early :) I have not adapted all tests in here yet